src/units/unit.cpp

```diff
@@ -1457,8 +1457,13 @@ static bool matches_ability_filter(const config & cfg, const std::string& tag_na
 	if(!bool_matches_if_present(filter, cfg, "affect_self", true))
 		return false;
 
-	if(!bool_matches_if_present(filter, cfg, "affect_allies", true))
+	if(!filter["affect_allies"].empty() && !cfg["affect_allies"].empty()){
+		if(!bool_matches_if_present(filter, cfg, "affect_allies", true)){
+			return false;
+		}
+	} else if(!filter["affect_allies"].empty() && filter["affect_allies"].str() != "empty"){
```

Comment:
This is very confusing. This should check if affect_allies exists in the filter at all. Then it should check if it is empty and return the comparison to what's in cfg or if it's not empty do the check as before.